pkg/api/types.go

```diff
@@ -172,6 +172,9 @@ type Volume struct {
 	// This is optional for now. If not specified, the Volume is implied to be an EmptyDir.
 	// This implied behavior is deprecated and will be removed in a future version.
 	VolumeSource `json:",inline,omitempty"`
+	// SecretName is the name of a Secret in the same namespace as the pod containing this volume
+	// used for volume authentication
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
SetSecrets([]secret)?  1 or many secrets seems easy enough.

What about this being a pattern for all volumes versus some volumes?  This is a satisfying solution, but only if it generally applies to all volumes.  If we want to restrict it to just those volumes that require secrets, it would require something like an AuthVolumePlugin interface.

Or is this not how we want to pair secrets and volumes at all?